packages/connection-storage/src/renderer.ts

```diff
@@ -44,17 +94,49 @@ export class ConnectionStorage {
   static get getLegacyConnections() {
     return this.ipc.getLegacyConnections;
   }
-  static get save() {
-    return this.ipc.save;
+  static async save({
+    connectionInfo,
+    signal,
+  }: {
+    signal?: AbortSignal;
+    connectionInfo: ConnectionInfo;
+  }): Promise<void> {
+    const result = await this.ipc.save({ connectionInfo, signal });
+    this.events.emit(ConnectionStorageEvents.ConnectionsChanged);
+    return result;
   }
-  static get delete() {
-    return this.ipc.delete;
+
+  static async delete({
+    id,
+    signal,
+  }: {
+    id: string;
+    signal?: AbortSignal;
+  }): Promise<void> {
+    const result = this.ipc.delete({ id, signal });
+    this.events.emit(ConnectionStorageEvents.ConnectionsChanged);
+    return result;
   }
+
   static get deserializeConnections() {
     return this.ipc.deserializeConnections;
   }
-  static get importConnections() {
-    return this.ipc.importConnections;
+  static async importConnections({
+    content,
+    options = {},
+    signal,
+  }: {
+    content: string;
+    options?: ImportConnectionOptions;
+    signal?: AbortSignal;
+  }): Promise<void> {
```

Comment:
The return is Promise<void> but we do return results. I guess we can't type them, right?

Reply:
I'm not sure what you mean. The current implementation of importConnections in connection-storage.ts already returns Promise<void>. I'm just propagating the return type. Do you have an example of what you mean?